### PR TITLE
SRE-30 → retire $wgCdnStylePath in favour of $wgResourceBasePath

### PR DESCRIPTION
**Pull request by ghost**
https://wikia-inc.atlassian.net/browse/SRE-30

It seems in all environments $wgCdnStylePath and $wgResourceBasePath have the same value, so let's stick to core MW's variable.

Also remove unused `$wgStyleSheetPath`

https://www.mediawiki.org/wiki/Manual:$wgResourceBasePath

https://github.com/Wikia/config/pull/2326

### CODE REVIEW

File: extensions/wikia/AssetsManager/tests/AssetsManagerSassBuilderTest.php [hunk 1]
```diff
@@ -9,30 +9,28 @@ class AssetsManagerSassBuilderTest extends WikiaBaseTest {
 
 	public function setUp() {
```
Comment (WikiaTech) [inline extensions/wikia/AssetsManager/tests/AssetsManagerSassBuilderTest.php:10]: MINOR Remove this method "setUp" to simply inherit it. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1185)

[project-id]: wikia:app

File: includes/wikia/GlobalFunctions.php [hunk 2]
```diff
@@ -111,27 +111,27 @@ function print_pre( $param, $return = 0 )
  */
 function wfReplaceImageServer( $url, $timestamp = false ) {
 	$wg = F::app()->wg;
```
Comment (WikiaTech) [inline includes/wikia/GlobalFunctions.php:113]: MAJOR Remove this unused "$wg" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

[project-id]: wikia:app

File: includes/wikia/GlobalFunctions.php [hunk 3]
```diff
@@ -111,27 +111,27 @@ function print_pre( $param, $return = 0 )
...
 			if ( $timestamp == "" ) {
 				$matches = array();
 				// @TODO: consider using wgStyleVersion
-				if ( 0 < preg_match( "/\/__cb([0-9]+)/i", $wg->CdnStylePath, $matches ) ) {
+				if ( 0 < preg_match( "/\/__cb([0-9]+)/i", $wgResourceBasePath, $matches ) ) {
 					$timestamp = $matches[1];
 				} else {
 					// This results in no caching of the image.  Bad bad bad, but the best way to fail.
-					Wikia::log( __METHOD__, "", "BAD FOR CACHING!: There is a call to " . __METHOD__ . " without a timestamp and we could not parse a fallback cache-busting number out of wgCdnStylePath.  This means the '{$url}' image won't be cacheable!" );
+					Wikia::log( __METHOD__, "", "BAD FOR CACHING!: There is a call to " . __METHOD__ . " without a timestamp and we could not parse a fallback cache-busting number out of wgResourceBasePath.  This means the '{$url}' image won't be cacheable!" );
```
Comment (WikiaTech) [inline includes/wikia/GlobalFunctions.php:129]: MAJOR Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)

[project-id]: wikia:app

File: includes/wikia/GlobalFunctions.php [hunk 4]
```diff
@@ -111,27 +111,27 @@ function print_pre( $param, $return = 0 )
...
 	wfDebug( __METHOD__ . ": requested url $url\n" );
 	if ( substr( strtolower( $url ), -4 ) != '.ogg' ) {
 		$url = str_replace( 'http://', 'https://',
 			str_replace( "//{$wgMedusaHostPrefix}images", '//' . str_replace( '.', '-', $wgMedusaHostPrefix ) . 'images', $url ) );
 		if ( strlen( $url ) > 8 && substr ( $url, 0, 8 ) == "https://" ) {
-			// If there is no timestamp, use the cache-busting number from wgCdnStylePath.
+			// If there is no timestamp, use the cache-busting number from wgResourceBasePath.
 			if ( $timestamp == "" ) {
 				$matches = array();
 				// @TODO: consider using wgStyleVersion
```
Comment (WikiaTech) [inline includes/wikia/GlobalFunctions.php:124]: INFO Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

[project-id]: wikia:app

File: extensions/wikia/AssetsManager/AssetsManager.class.php [hunk 5]
```diff
@@ -46,8 +46,8 @@ private function loadConfig() {
 	 */
 	public static function getInstance() {
 		if( self::$mInstance == false ) {
```
Comment (WikiaTech) [inline extensions/wikia/AssetsManager/AssetsManager.class.php:48]: MINOR Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

[project-id]: wikia:app

File: includes/wikia/services/sass/Filter/CdnRewriteFilter.class.php [hunk 6]
```diff
@@ -20,14 +20,14 @@ public function __construct( $cdnUrl ) {
 	public function process( $contents ) {
 		wfProfileIn(__METHOD__);
 
-		if (strpos($contents, "wgCdnStylePath") !== false) { // faster to skip the regex in most cases
+		if (strpos($contents, "wgResourceBasePath") !== false) { // faster to skip the regex in most cases
```
Comment (WikiaTech) [inline includes/wikia/services/sass/Filter/CdnRewriteFilter.class.php:23]: INFO Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app

File: includes/wikia/services/sass/Filter/CdnRewriteFilter.class.php [hunk 7]
```diff
@@ -20,14 +20,14 @@ public function __construct( $cdnUrl ) {
...
 			// Because of fonts in CSS, we have to allow for lines with multiple url()s in them.
 			// This will rewrite all but the last URL on the line (the last regex will fix the final URL and remove the special comment).
 			$wasChanged = true;
 
 			// TODO: refactor?
```
Comment (WikiaTech) [inline includes/wikia/services/sass/Filter/CdnRewriteFilter.class.php:28]: INFO Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

[project-id]: wikia:app

**Comment by WikiaTech** [comment]
SonarQube analysis reported 180 issues
* ![MAJOR][MAJOR] 109 major
* ![MINOR][MINOR] 24 minor
* ![INFO][INFO] 47 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AssetsManager.class.php#L7](https://github.com/Wikia/app/blob/81278477c0bd8a0734ef711e3c25c8a88ef16d7a/extensions/wikia/AssetsManager/AssetsManager.class.php#L7): Class "AssetsManager" has 31 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L55](https://github.com/Wikia/app/blob/81278477c0bd8a0734ef711e3c25c8a88ef16d7a/extensions/wikia/AssetsManager/AssetsManager.class.php#L55): Don't use an OnMakeGlobalVariablesScript hook handler to register JS variables. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AOnMakeGlobalVariablesScript)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L90](https://github.com/Wikia/app/blob/81278477c0bd8a0734ef711e3c25c8a88ef16d7a/extensions/wikia/AssetsManager/AssetsManager.class.php#L90): The Cyclomatic Complexity of this function "getURL" is 21 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L122](https://github.com/Wikia/app/blob/81278477c0bd8a0734ef711e3c25c8a88ef16d7a/extensions/wikia/AssetsManager/AssetsManager.class.php#L122): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L180](https://github.com/Wikia/app/blob/81278477c0bd8a0734ef711e3c25c8a88ef16d7a/extensions/wikia/AssetsManager/AssetsManager.class.php#L180): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L396](https://github.com/Wikia/app/blob/81278477c0bd8a0734ef711e3c25c8a88ef16d7a/extensions/wikia/AssetsManager/AssetsManager.class.php#L396): Remove the unused function parameter "$minify". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L405](https://github.com/Wikia/app/blob/81278477c0bd8a0734ef711e3c25c8a88ef16d7a/extensions/wikia/AssetsManager/AssetsManager.class.php#L405): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L765](https://github.com/Wikia/app/blob/81278477c0bd8a0734ef711e3c25c8a88ef16d7a/extensions/wikia/AssetsManager/AssetsManager.class.php#L765): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L799](https://github.com/Wikia/app/blob/81278477c0bd8a0734ef711e3c25c8a88ef16d7a/extensions/wikia/AssetsManager/AssetsManager.class.php#L799): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L799](https://github.com/Wikia/app/blob/81278477c0bd8a0734ef711e3c25c8a88ef16d7a/extensions/wikia/AssetsManager/AssetsManager.class.php#L799): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 276 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 147 major
* ![MINOR][MINOR] 24 minor
* ![INFO][INFO] 102 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L1047](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/includes/DefaultSettings.php#L1047): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L1183](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/includes/DefaultSettings.php#L1183): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L5514](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/includes/DefaultSettings.php#L5514): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L7](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L7): Class "AssetsManager" has 31 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L55](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L55): Don't use an OnMakeGlobalVariablesScript hook handler to register JS variables. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AOnMakeGlobalVariablesScript)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L90](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L90): The Cyclomatic Complexity of this function "getURL" is 21 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L122](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L122): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L180](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L180): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L396](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L396): Remove the unused function parameter "$minify". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L405](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L405): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

[project-id]: wikia:app
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by mszabo-wikia — APPROVED** [review]

File: extensions/wikia/AchievementsII/css/leaderboard_oasis.scss [hunk 8]
```diff
@@ -67,13 +67,13 @@
...
-			background: url(/extensions/wikia/AchievementsII/images/bronze_badge_blank.png); /* $wgCdnStylePath */
+			background: url(/extensions/wikia/AchievementsII/images/bronze_badge_blank.png); /* $wgResourceBasePath */
 		}
 		.silver span {
-			background: url(/extensions/wikia/AchievementsII/images/silver_badge_blank.png); /* $wgCdnStylePath */
+			background: url(/extensions/wikia/AchievementsII/images/silver_badge_blank.png); /* $wgResourceBasePath */
 		}
 		.gold span {
-			background: url(/extensions/wikia/AchievementsII/images/gold_badge_blank.png); /* $wgCdnStylePath */
+			background: url(/extensions/wikia/AchievementsII/images/gold_badge_blank.png); /* $wgResourceBasePath */
```
Comment (macbre) [inline extensions/wikia/AchievementsII/css/leaderboard_oasis.scss:76]: I'd double check if these comments are actually still required to perform a URL rewrite on AssetsManager SASS renderer.

**Review by macbre — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 295 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 151 major
* ![MINOR][MINOR] 24 minor
* ![INFO][INFO] 117 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L1047](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/includes/DefaultSettings.php#L1047): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L1183](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/includes/DefaultSettings.php#L1183): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L5514](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/includes/DefaultSettings.php#L5514): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L7](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L7): Class "AssetsManager" has 31 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L55](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L55): Don't use an OnMakeGlobalVariablesScript hook handler to register JS variables. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AOnMakeGlobalVariablesScript)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L90](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L90): The Cyclomatic Complexity of this function "getURL" is 21 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L122](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L122): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L180](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L180): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L396](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L396): Remove the unused function parameter "$minify". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AssetsManager.class.php#L405](https://github.com/Wikia/app/blob/f67eb71ba06e96bc39ca3313300f551b9c412e99/extensions/wikia/AssetsManager/AssetsManager.class.php#L405): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

[project-id]: wikia:app
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**